src/htm/engine/RESTapi.cpp

```diff
@@ -95,10 +95,14 @@ std::string RESTapi::put_input_request(const std::string &id,
     NTA_CHECK(itr != resource_.end()) << "Context for resource '" + id + "' not found.";
     itr->second.t = time(0);
 
-    Array a;
-    a.fromJSON(data);
-
-    itr->second.net->getRegion(region_name)->setInputData(input_name, a);
+    //Array a;
+    //a.fromJSON(data);
+    //itr->second.net->setInputData(input_name, a);
+
+    // Topic of PR #585
+    // The above call would not work because we don't have the type info for a.
+    // lets use itr->second.net->setInputData(input_name, vm);  It knows the type.
```

Comment:
TODO remove later in #834 (I assume its #834, rather than #585 ?)

Reply:
Sorry, it was #858.   Can you tell I am very dyslexic?  That is also why my spelling is so bad.